js/components/graph/Graph.js

```diff
@@ -397,25 +469,45 @@ export default class Graph extends React.Component {
     );
   };
 
-  incrementZoom = (increase, zoomFactorRate) => {
-    // onButtonRelease calls are required when a button becomes disabled
-    // because it loses its ability to detect mouseUp event
-    if (increase) {
-      if (this.state.zoomFactor > 0.5) {
-        // zooming allowed
-        this.setState({ zoomFactor: this.state.zoomFactor - zoomFactorRate });
-      } else {
-        // button becomes disabled
-        this.onButtonRelease();
-      }
+  /** Zoom into the graph by calculating new viewbox dimensions
+   *
+   * @param {number} zoomMode - Determines whether to zoom in, zoom out, or rerender at current zoom level
+   */
+  zoomViewbox = (zoomMode) => {
+    let containerWidth = 0;
+    let containerHeight = 0;
+
+    if (document.getElementById("react-graph") !== null){
+      var reactGraph = document.getElementById("react-graph");
+      containerWidth = reactGraph.clientWidth;
+      containerHeight = reactGraph.clientHeight;
+    }
+
+    var newZoomFactor = this.state.zoomFactor;
+    if (zoomMode === this.zoomEnum.ZOOM_IN){
+      newZoomFactor -= ZOOM_INCREMENT;
+    } else if (zoomMode === this.zoomEnum.ZOOM_OUT) {
+      newZoomFactor += ZOOM_INCREMENT;
+    } else if (zoomMode === this.zoomEnum.RESET_ZOOM){
+      newZoomFactor = 1
+    }
+    let newViewboxWidth = this.state.width;
+    let newViewboxHeight = this.state.height;
+    if (document.getElementById("generateRoot")){
```

Comment:
The special case is there so that the graph renders in the middle in **Generate** mode. Since the fix that allows **Generate** mode to run is in a different PR, I can't test any better solutions.

Should we keep the code from the else branch for now, and then fix any issues that arise in **Generate** once the Generate fix PR is merged in, so we can test while developing?

Reply:
The generate fix has been merged so try pulling now

Reply:
Yes that's interesting, I'll revise my original comment. Why don't you always use the logic in the generate mode (taking the max of the width and container width, and same for height), even for the graph page. Hopefully this should make the smaller graphs on the graph page centered as well.

Also, I just pulled in #1216, so if all of you pull from master, you'll get those fixes!

Reply:
I tested this and there is a bug here. As is, in Generate, when you try zooming it "jumps" to a suddenly larger size, and does not return to the original size properly when reset. This is because the initial value of `viewBoxDim.width` is not set to the maximum of data.width and containerWidth, yet it is used anyway on the first render.

Reply:
If we always use the logic in generate mode (taking max of width & container width), the graph page also renders in the middle, like so. Is this fine?
![image](https://user-images.githubusercontent.com/22108651/122274986-2f1e9a00-ceb1-11eb-99d0-bed5ac0817f4.png)

Reply:
@kenneth-miura hmmm yeah, it will also depend on screen size/resolution. There's a deeper issue here, so for now I'm okay leaving the code as-is, with the separate logic for the generate page.